drivers/video/hm01b0.c

```diff
@@ -0,0 +1,308 @@
+{
+	struct hm01b0_data *data = dev->data;
+
+	*fmt = data->fmt;
+	LOG_DBG("HM01B0 get_fmt: %d x %d, fmt: %s", fmt->width, fmt->height,
+		VIDEO_FOURCC_TO_STR(fmt->pixelformat));
+	return 0;
+}
+
+static int hm01b0_set_stream(const struct device *dev, bool enable, enum video_buf_type type)
+{
+	const struct hm01b0_config *config = dev->config;
+
+	/* SET MODE_SELECT */
+	return video_write_cci_reg(&config->i2c, HM01B0_CCI_STS, enable ? 1 : 0);
+}
+
+static int hm01b0_soft_reset(const struct device *dev)
+{
+	const struct hm01b0_config *config = dev->config;
+	uint32_t val = 0xff;
+	int ret;
+
+	ret = video_write_cci_reg(&config->i2c, HM01B0_CCI_RESET, 0x01);
+	if (ret != 0) {
+		LOG_ERR("Error writing HM01B0_CCI_RESET (%d)", ret);
+		return ret;
+	}
+
+	for (int retries = 0; retries < 10; retries++) {
+		ret = video_read_cci_reg(&config->i2c, HM01B0_CCI_STS, &val);
+		if (ret != 0 || val == 0x0) {
+			break;
+		}
+		k_msleep(100);
+	}
+	if (ret != 0) {
+		LOG_ERR("Soft reset error (%d)", ret);
+	}
+	return ret;
+}
+
+static DEVICE_API(video, hm01b0_driver_api) = {
+	.set_format = hm01b0_set_fmt,
+	.get_format = hm01b0_get_fmt,
+	.set_stream = hm01b0_set_stream,
+	.get_caps = hm01b0_get_caps,
+};
+
+static bool hm01b0_check_connection(const struct device *dev)
+{
+	const struct hm01b0_config *config = dev->config;
+	uint32_t model_id;
+	int ret;
+
+	ret = video_read_cci_reg(&config->i2c, HM01B0_CCI_ID, &model_id);
+	if (ret != 0) {
+		LOG_ERR("Error reading id reg (%d)", ret);
+		return false;
+	}
+	return (model_id == HM01B0_ID);
+}
+
+static int hm01b0_init(const struct device *dev)
+{
+	struct hm01b0_data *data = dev->data;
+	const struct hm01b0_config *config = dev->config;
+	int ret;
+
+	if (config->data_bits == 8) {
+		data->ctrl_val = 0x02;
+	} else if (config->data_bits == 4) {
+		data->ctrl_val = 0x42;
+	} else if (config->data_bits == 1) {
+		data->ctrl_val = 0x22;
+	} else {
+		LOG_ERR("Invalid data bits!");
+		return -ENODEV;
+	}
+
+	if (!hm01b0_check_connection(dev)) {
+		LOG_ERR("%s is not ready", dev->name);
+		return -ENODEV;
+	}
+
+	ret = hm01b0_soft_reset(dev);
+	if (ret != 0) {
+		LOG_ERR("error sof reset (%d)", ret);
+		return ret;
+	}
+
+	struct video_format fmt = {
+		.pixelformat = VIDEO_PIX_FMT_GREY,
+		.width = 160,
+		.height = 120,
+		.type = VIDEO_BUF_TYPE_OUTPUT,
+		.pitch = 160 * video_bits_per_pixel(VIDEO_PIX_FMT_GREY) / BITS_PER_BYTE,
+	};
+
+	ret = hm01b0_set_fmt(dev, &fmt);
+	if (ret != 0) {
+		LOG_ERR("Error setting video format (%d)", ret);
+		return ret;
+	}
+	return 0;
+}
+
+#define HM01B0_INIT(inst)                                                                          \
+	const struct hm01b0_config hm01b0_config_##inst = {                                        \
+		.i2c = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.data_bits = DT_INST_PROP(inst, data_bits),                                        \
```

Comment:
Instead of the runtime if statement you could shrink the driver flash usage a small amount with:
```suggestion
		.ctrl_val = HM01B0_CTRL_VAL(DT_INST_PROP(inst, data_bits)),      \
```

And:

```
#define HM01B0_CTRL_VAL(data_bits) \
    ((data_bits) == 8 ? 0x02 : \
     (data_bits) == 4 ? 0x42 : \
     (data_bits) == 1 ? 0x22 : 0x00)
```

The Devicetree enum will ensure only 1, 4 or 8 is valid for the `data-bits` property.